core/priority/calculate.go

```diff
@@ -37,28 +41,40 @@ func calculateResult(msgs []*pbv1.PriorityMsg, minRequired int) (*pbv1.PriorityR
 	}
 
 	// Group all priority sets by topic
-	prioritySetsByTopic := make(map[string][][]string)
+	proposalsByTopic := make(map[[32]byte][]*pbv1.PriorityTopicProposal)
 	for _, msg := range sortInput(msgs) {
 		for _, topic := range msg.Topics {
-			prioritySetsByTopic[topic.Topic] = append(prioritySetsByTopic[topic.Topic], topic.Priorities)
+			topic := topic // Copy iteration value since it is a pointer
+			topicHash, err := hashProto(topic.Topic)
+			if err != nil {
+				return nil, err
+			}
+			proposalsByTopic[topicHash] = append(proposalsByTopic[topicHash], topic)
 		}
 	}
 
 	// Calculate cluster wide resulting priorities by topic.
 	var topicResults []*pbv1.PriorityTopicResult
-	for topic, prioritySet := range prioritySetsByTopic {
+	for _, proposals := range proposalsByTopic {
 		// Calculate overall score for all priorities in the topic
 		// which effectively orders by count then by overall priority.
 		var (
-			scores        = make(map[string]int)
-			allPriorities []string
+			scores        = make(map[[32]byte]int)
```

Comment:
can we give `[32]byte` a type? will make it more readable

Reply:
yeah not sure, I actually removed the `instanceKey` type, reverting to just `[32]byte`. I think defining types is mostly useful for the scope of the type is very large. In this case, it is pretty small.